client/my-sites/backup/backup-contents-page/file-browser/file-browser-node.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button, Icon } from '@wordpress/components';
+import { chevronDown, chevronRight } from '@wordpress/icons';
+import { FunctionComponent, useCallback, useState } from 'react';
+import FileTypeIcon from './file-type-icon';
+import { FileBrowserItem } from './types';
+import { useBackupContentsQuery } from './use-backup-contents-query';
+
+interface FileBrowserNodeProps {
+	item: FileBrowserItem;
+	path: string;
+	siteId: number;
+	rewindId: number;
+}
+
+const FileBrowserNode: FunctionComponent< FileBrowserNodeProps > = ( {
+	item,
+	path,
+	siteId,
+	rewindId,
+} ) => {
+	const isRoot = path === '/';
+	const isDirectory = item.type === 'dir';
+
+	const [ fetchContentsOnMount, setFetchContentsOnMount ] = useState< boolean >( isRoot );
+	const [ isOpen, setIsOpen ] = useState< boolean >( isRoot );
+
+	const {
+		isSuccess,
+		isInitialLoading,
+		data: backupFiles,
+	} = useBackupContentsQuery( siteId, rewindId, path, fetchContentsOnMount );
+
+	const handleClick = useCallback( () => {
+		if ( ! isDirectory ) {
+			return;
+		}
+
+		if ( ! isOpen ) {
+			setFetchContentsOnMount( true );
+			setIsOpen( true );
+		}
+
+		setIsOpen( ! isOpen );
+	}, [ isDirectory, isOpen ] );
+
+	const renderChildren = () => {
+		if ( isInitialLoading ) {
+			return <div className="file-browser-node__loading placeholder" />;
+		}
+
+		// @TODO: Add a message when the API fails to fetch
```

Comment:
The public-api will return an empty array on errors but might error / timeout itself.  We can potentially pass through errors from the VP API if it's useful.